addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRuleTest.java

```diff
@@ -0,0 +1,312 @@
+        assertThat(httpMessagesSent, hasSize(1));
+        Alert alert = alertsRaised.get(0);
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+        assertEquals(rule.getName(), alert.getName());
+        assertEquals(rule.getWascId(), alert.getWascId());
+    }
+
+    @Test
+    void shouldSendGetRequestWhenOriginalRequestWasNotGet() throws HttpMalformedHeaderException {
+        // Given
+        String path = "";
+        HttpMessage msg = getHttpMessage(path);
+        msg.getRequestHeader().setMethod(HttpRequestHeader.POST);
+        msg.getRequestHeader().addHeader(HttpHeader.CONTENT_TYPE, ACTUATOR_CONTENT_TYPE);
+        msg.setRequestBody("field1=value1&field2=value2");
+        rule.init(msg, parent);
+
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent, hasSize(2));
+        assertEquals(HttpRequestHeader.GET, httpMessagesSent.get(0).getRequestHeader().getMethod());
+        assertEquals(0, httpMessagesSent.get(0).getRequestBody().length());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {ACTUATOR_CONTENT_TYPE, "Content-Type: application/json"})
+    void shouldAlertIfTestedUrlRespondsOKWithRelevantContent(String contentType)
+            throws HttpMalformedHeaderException {
+        // Given
+        String servePath = "/shouldAlert";
+        String alertPath = REQ_PATH;
+        this.nano.addHandler(
+                new NanoServerHandler(servePath) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, NanoHTTPD.MIME_HTML, IRRELEVANT_BODY);
+                    }
+                });
+        this.nano.addHandler(
+                new NanoServerHandler("/" + alertPath) {
+                    @Override
+                    protected Response serve(NanoHTTPD.IHTTPSession session) {
+                        return newFixedLengthResponse(
+                                Response.Status.OK, contentType, RELEVANT_BODY);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(servePath);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertThat(httpMessagesSent, hasSize(1));
+        Alert alert = alertsRaised.get(0);
+        assertEquals(Alert.RISK_MEDIUM, alert.getRisk());
+        assertEquals(Alert.CONFIDENCE_MEDIUM, alert.getConfidence());
+        assertEquals(rule.getName(), alert.getName());
+        assertEquals(rule.getWascId(), alert.getWascId());
+    }
+
+    @Test
+    void shouldNotAlertIfTestedUrlRespondsOKWithIrrelevantContent()
+            throws HttpMalformedHeaderException {
+        // Given
+        String servePath = "/shouldAlert";
```

Comment:
Was this called `shouldAlert` on purpose? The test method is asserting the opposite (same for following test).